### PR TITLE
Add initial iteration of frontend AMI filter

### PR DESCRIPTION
**Pull request by avaleske**
## Issue 515

- Addresses #515
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Wire up AMI filter on the frontend, with a simple dropdown for now.

Also changes the ami filter name from `ami` to `minAmiPercentage` for clarity. (cc @abbiefarr for visibility)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Run the tests, try adding the ami filter on the frontend.

- [ ] Desktop View
- [ ] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: sites/public/pages/listings.tsx [hunk 1]
```diff
@@ -276,6 +296,17 @@ const ListingsPage = () => {
...
+            {/* TODO(#515): Add more explanation and an ami percentage
+            calculator to this filter */}
+            <Select
+              id="amiSelect"
+              name={FrontendListingFilterStateKeys.minAmiPercentage}
+              label={t("listingFilters.minAmiPercentageLabel")}
+              register={register}
+              controlClassName="control"
+              options={amiOptions}
+              defaultValue={filterState?.minAmiPercentage?.toString()}
```
Comment (willrlin) [inline sites/public/pages/listings.tsx:308]: Is the `toString` necessary here?

File: ui-components/src/locales/bn.json [hunk 2]
```diff
@@ -1356,7 +1356,26 @@
         "noResultsSubtitle": "সেরা ফলাফলের জন্য ফিল্টার সম্পাদনা বা অপসারণ করুন।",
         "includeUnknowns": "অসম্পূর্ণ তথ্য সহ ফলাফল অন্তর্ভুক্ত করুন",
         "senior": "সিনিয়র হাউজিং (62+)",
-        "specialNeeds": "বিশেষ চাহিদা এবং অক্ষমতা"
+        "specialNeeds": "বিশেষ চাহিদা এবং অক্ষমতা",
+        "minAmiPercentageLabel": "যাদের আয় আছে তাদের জন্য ইউনিট আলাদা রাখা হয়েছে",
+        "minAmiPercentageOptions": {
+          "amiOption20": "এএমআই এর 20% বা তার বেশি",
+          "amiOption25": "25% AMI বা তার বেশি",
+          "amiOption30": "এএমআই এর 30% বা তার বেশি",
```
Comment (willrlin) [inline ui-components/src/locales/bn.json:1364]: Just curious if you know why some of these have `25% AMI` and others don't actually mention it?

**Review by willrlin — APPROVED** [review]

**Review by abbiefarr — APPROVED** [review]
Sorry about the merge conflicts! They shouldn't be too big a change.

File: ui-components/src/locales/bn.json (hunk 2 above)
Comment (avaleske) [reply ui-components/src/locales/bn.json:1364]: I just copied from google translate so I have no idea.

File: sites/public/pages/listings.tsx (hunk 1 above)
Comment (avaleske) [reply sites/public/pages/listings.tsx]: Yeah, the type of `defaultValue` is `string` not `string | number`

**Pull request merged**